### DISCUSSION TITLE
onafterstatechange get all foms fields data

### DISCUSSION THREAD

**Discussion by ajaztanoli** [filamentphp/filament#15172] [General]
Hi everyone,
i am newbie in laraver filament i am having issue getting all fields data on afterStateUpdated fields  plz can anyone assisst me here.

class AddOverTimeResource extends Resource
{
      
    {   
        
        return $form
            ->schema([
                Select::make('overtime_sheet_id')
                    ->label('Select Overtime')
                    ->options(function (): array {
                        return overtime_sheet::all()->pluck('drillment_name', 'id')->all();
                    })
                    ->required()
                    ->reactive()
                    ->afterStateUpdated(function ($state, callable $set,Get $get) {
                            if ($state) {
                            $overtimesheet = overtime_sheet::find($state);
                                if ($overtimesheet) {
                                // Set the total_overtime_hours field globally
                                $set('total_overtime_hours', $overtimesheet->total_hours);
                            } else {
                                $set('total_overtime_hours', 0);
                            }
                        } else {
                            $set('total_overtime_hours', 0);
                        }
                    }),
    
                TextInput::make('total_overtime_hours')
                    ->disabled(),  // Disable if auto-calculated
    
                // Repeater for adding employees' overtime
                Repeater::make('employees_overtime')
                    ->label('Add User')
                    ->schema([
                        Select::make('user_id')
                            ->label('Select User')
                            ->options(function (): array {
                                return employees::all()->pluck('name', 'id')->all();
                            })
                            ->required()
                            ->reactive()
                            ->afterStateUpdated(function ($state, callable $set, callable $get) {
                                // Fetch the total_overtime_hours value before performing calculations
                                $formState = $get(); // This will give you the state of the whole form
                                  dd($formState);
                                  // $totalOvertime = $get('total_overtime_hours');  // Get total_overtime_hours from the form state
                               // dd(vars:$totalOvertime);
                                // Ensure we are working with a valid number
                                if ($totalOvertime === null) {
                                    $totalOvertime = 0;  // Default value if not set
                                }
    
                                if ($state) {
                                    $employee = employees::find($state);
                                    if ($employee) {
                                        // Calculate Total Overtime based on employee's basic pay
                                        $totalOvertimeAmount = $totalOvertime * $employee->basic_pay;
                                        $set('Total_over_time', $totalOvertimeAmount);  // Set the calculated total overtime
                                    } else {
                                        $set('Total_over_time', 0);  // Default if employee is not found
                                    }
                                } else {
                                    $set('Total_over_time', 0);  // Default if no user is selected
                                }
                            }),
    
                        TextInput::make('Total_over_time')->disabled(),  // Disable if auto-calculated
                    ])
                    ->reactive()
                    ->columns(2),
            ]);
  
    }
  
}
also i am attaching image what i want to acchive .
![Create-Overtime](https://github.com/user-attachments/assets/ef348b03-cb5b-4090-aa37-60deab4dfbb6)

**Comment by jinbatsu** [comment]
From repeater, we can use `../` to get the parent field.
ex.: `$get('../../total_overtime_hours')`
https://filamentphp.com/docs/3.x/forms/fields/repeater#using-get-to-access-parent-field-values

**Reply by ajaztanoli** [reply]
@jinbatsu Thank you its working for me!
